src/runtime_src/driver/xclng/drm/xocl/subdev/icap.c

```diff
@@ -2021,21 +2024,19 @@ static int icap_lock_bitstream(struct platform_device *pdev, const xuid_t *id,
 	mutex_lock(&icap->icap_lock);
 
 	if (ICAP_PRIVILEGED(icap)) {
-		if (!icap_bitstream_in_use(icap, pid) ||
-			uuid_equal(id, &icap->icap_bitstream_uuid)) {
-			uuid_copy(&icap->icap_bitstream_uuid, id);
+		if (uuid_equal(id, &icap->icap_bitstream_uuid))
```

Comment:
Yeah, you don't want to change ID while locking/unlocking it. The original code looks very strange. Can't remember what I was thinking back then, but thanks for fixing it :-).